tensorflow/stream_executor/rocm/rocm_driver.cc

```diff
@@ -0,0 +1,1356 @@
+          << " bdx: " << block_dim_x << " bdy: " << block_dim_y
+          << " bdz: " << block_dim_z << " smem: " << shared_mem_bytes;
+  hipError_t res = hipModuleLaunchKernel(
+     function, grid_dim_x, grid_dim_y, grid_dim_z, block_dim_x, block_dim_y,
+     block_dim_z, shared_mem_bytes, stream, kernel_params, extra);
+  if (res != hipSuccess) {
+    LOG(ERROR) << "failed to launch ROCM kernel: " << function
+               << "; result: " << ToString(res);
+    return false;
+  }
+  VLOG(2) << "successfully launched kernel";
+  return true;
+}
+
+/* static */ bool GpuDriver::LoadPtx(GpuContext* context,
+				     const char* ptx_contents,
+				     hipModule_t* module) {
+  LOG(ERROR) << "Feature not supported on ROCm platform (LoadPtx)";
+  return false;
+}
+  
+/* static */ port::Status GpuDriver::LoadCubin(GpuContext* context,
+					       const char* cubin_bytes,
+					       hipModule_t* module) {
+  return port::Status{
+    port::error::INTERNAL,
+    "Feature not supported on ROCm platform (LoadCubin)"};
+}
+  
+/* static */ bool GpuDriver::LoadHsaco(GpuContext* context,
+				       const char* hsaco_contents,
+				       hipModule_t* module) {
+  port::Notification notification;
+  bool ret = true;
+  GetDriverExecutor()->Schedule(
+      [context, hsaco_contents, module, &ret, &notification]() {
+        ScopedActivateContext activation{context};
+        void* hsaco_data = const_cast<char*>(hsaco_contents);
+
+        hipError_t res = hipModuleLoadData(module, hsaco_data);
+
+        if (res != hipSuccess) {
+          LOG(ERROR) << "failed to load HSACO: " << ToString(res);
+          ret = false;
+          notification.Notify();
+        }
+
+        CHECK(module != nullptr);
+        notification.Notify();
+      });
+  notification.WaitForNotification();
+
+  return ret;
+}
+
+/* static */ bool GpuDriver::SynchronousMemsetUint8(GpuContext* context,
+                                                     hipDeviceptr_t location,
+                                                     uint8 value, size_t size) {
+  ScopedActivateContext activation{context};
+  hipError_t res = hipMemset(location, value, size);
+  if (res != hipSuccess) {
+    LOG(ERROR) << "failed to memset memory: " << ToString(res);
+    return false;
+  }
+  return true;
+}
+
+/* static */ bool GpuDriver::SynchronousMemsetUint32(GpuContext* context,
+                                                      hipDeviceptr_t location,
+                                                      uint32 value,
+                                                      size_t uint32_count) {
+  ScopedActivateContext activation{context};
+  void * pointer = absl::bit_cast<void *>(location);
+  unsigned char valueC = static_cast<unsigned char>(value);
+  uint32_t value32 = (valueC << 24) | (valueC << 16) | (valueC << 8) | (valueC) ;
+  assert (value32 == value); // if mismatch this indicates case where hipMemsetAsyc can't emulate hipMemSetD32
```

Comment:
`if (value32 != value) {
  LOG(ERROR) << "...";
  return false;
}`?

Reply:
will be fixed in the next commit